File: lib/src/main/java/com/qwwuyu/lib/gson/IgnoreTypeAdapter.java
package com.qwwuyu.lib.gson;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;

class IgnoreTypeAdapter extends TypeAdapter<Ignore> {
    @Override
    public void write(final JsonWriter out, final Ignore value) throws IOException {
        out.nullValue();
    }

    @Override
    public Ignore read(final JsonReader in) throws IOException {
        in.skipValue();
        return Ignore.single;
    }
}

File: lib/src/main/kotlin/com/qwwuyu/lib/gson/IgnoreTypeAdapter.kt
package com.qwwuyu.lib.gson

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import java.io.IOException

internal class IgnoreTypeAdapter : TypeAdapter<Ignore?>() {
    @Throws(IOException::class)
    override fun write(out: JsonWriter, value: Ignore?) {
        out.nullValue()
    }

    @Throws(IOException::class)
    override fun read(`in`: JsonReader): Ignore {
        `in`.skipValue()
        return Ignore.single
    }
}